sw/device/lib/dif/dif_rstmgr.c

```diff
@@ -287,3 +287,13 @@ dif_result_t dif_rstmgr_software_reset_is_held(
 
   return kDifOk;
 }
+
+dif_result_t dif_rstmgr_software_device_reset(const dif_rstmgr_t *handle) {
+  if (handle == NULL) {
+    return kDifBadArg;
+  }
+
+  mmio_region_write32(handle->base_addr, RSTMGR_RESET_REQ_REG_OFFSET, 0xa);
```

Comment:
nit: maybe just drop a TODO here to replace `0xa` with the appropriate enum when #8663 merges.